docs/reference/transform/checkpoints.asciidoc

```diff
@@ -5,50 +5,70 @@
-indices. This check is done based on the interval defined in the transform's 
+Using a simple periodic timer, the {transform} checks for changes to the source
+indices. This check is done based on the interval defined in the transform's
 `frequency` property.
 +
 If the source indices remain unchanged or if a checkpoint is already in progress
 then it waits for the next timer.
 
-. Identifies which entities have changed.
+If changes are found a checkpoint gets created.
+
+. Identifies which entities or time buckets have changed.
 +
-The {transform} searches to see which entities have changed since the last time 
-it checked. The `sync` configuration object in the {transform} identifies a time 
-field in the source indices. The {transform} uses the values in that field to 
-synchronize the source and destination indices.
- 
-. Updates the destination index (the {dataframe}) with the changed entities.
+The {transform} searches to see which entities or time buckets have changed
+between the last and the new checkpoint. The {transform} uses the values to
+synchronize the source and destination indices with fewer operations than a
+full re-run.
+
+. Updates the destination index (the {dataframe}) with the changes
 +
 --
-The {transform} applies changes related to either new or changed entities to the 
-destination index. The set of changed entities is paginated. For each page, the 
-{transform} performs a composite aggregation using a `terms` query. After all 
-the pages of changes have been applied, the checkpoint is complete.
+The {transform} applies changes related to either new or changed entities or
+time buckets to the destination index. The set of changes can be paginated. The
+{transform} performs a composite aggregation like for the run once case, however
+injects query filters based on step 2, to reduce the amount work. After all
+changes have been applied, the checkpoint is complete.
 --
 
 This checkpoint process involves both search and indexing activity on the
 cluster. We have attempted to favor control over performance while developing
-{transforms}. We decided it was preferable for the {transform} to take longer to 
-complete, rather than to finish quickly and take precedence in resource 
-consumption. That being said, the cluster still requires enough resources to 
-support both the composite aggregation search and the indexing of its results. 
+{transforms}. We decided it was preferable for the {transform} to take longer to
+complete, rather than to finish quickly and take precedence in resource
+consumption. That being said, the cluster still requires enough resources to
+support both the composite aggregation search and the indexing of its results.
 
 TIP: If the cluster experiences unsuitable performance degradation due to the
 {transform}, stop the {transform} and refer to <<transform-performance>>.
 
+[discrete]
+[[ml-transform-checkpoint-heuristics]]
+== Change Detection Heuristics
```

Comment:
I created some visuals about change point detection, e.g. in https://github.com/elastic/elasticsearch/pull/63315

This is a very technical one, but maybe we can take the ideas from it to create a better visual for this place.